### PR TITLE
add max_output_per_channel tests

### PR DESCRIPTION
**Pull request by assafgi**

### CODE REVIEW

File: tests/integration/test_sanity.py [hunk 1]
```diff
@@ -27,7 +18,19 @@ def test_echo_hello_command(self):
...
+    def test_max_output_exceeds_maximum(self):
         cmd = self.client.run('MyHostId', 'bash', '-ce', 'yes')
         with self.assertRaises(CommandAbortedByOverflow):
             cmd.result()
+
+    def test_max_output_per_channel_set(self):
+        max_output_per_channel = 3
+        for val, expected_ret in [('123', '0'), ('1234', 'overflowed')]:
+            cmd = self.client.run(
+                'MyHostId', 'bash', '-ce', 'python -c "print(\'{}\', end=\'\')"'.format(val),
```
Comment (DoronCohen) [inline tests/integration/test_sanity.py]: you acn use `echo` for simplicity. it has a flag for not printing endline (`-n` I think)

File: tests/integration/test_sanity.py [hunk 2]
```diff
@@ -27,7 +18,19 @@ def test_echo_hello_command(self):
...
+                max_output_per_channel=max_output_per_channel)
+            ret = get_retcode(self.client.redis, cmd.job_id)
+            self.assertEqual(ret, expected_ret)
+            if expected_ret == 0:
```
Comment (DoronCohen) [inline tests/integration/test_sanity.py]: do we want to validate the output when it's overflowed?

File: tests/uts/test_agent.py [hunk 3]
```diff
@@ -134,3 +136,35 @@ def test_last_exception(self):
...
+            ('1\\n', 1, 3, '0'),
+            ('12\\n', 1, 3, '0'),
+            ('1234', 1, 3, 'overflowed'),
+            ('123\\n', 1, 3, 'overflowed'),
+            ('12\\n3', 1, 3, 'overflowed'),
+            ('1', MAX_OUTPUT_PER_CHANNEL, MAX_OUTPUT_PER_CHANNEL, '0'),
+            ('1', MAX_OUTPUT_PER_CHANNEL, None, '0'),
+            ('1', MAX_OUTPUT_PER_CHANNEL + 1, MAX_OUTPUT_PER_CHANNEL, 'overflowed'),
+            ('1', MAX_OUTPUT_PER_CHANNEL + 1, None, 'overflowed'),
+            ('1', MAX_OUTPUT_PER_CHANNEL + 1, MAX_OUTPUT_PER_CHANNEL + 1, '0'),
```
Comment (DoronCohen) [inline tests/uts/test_agent.py]: nice. next time check out `namedtuple` for a more readable code

File: tests/uts/test_client.py [hunk 4]
```diff
@@ -82,20 +82,25 @@ def get_command_from_redis(self) -> dict:
...
-            'cmd': ['bash', '-ce', 'echo hello'],
-            'line_timeout': None,
-            'log_file': None,
-            'max_output_per_channel': 10485760,
-            'set_new_logpath': None,
-            'timeout': 3600.0}
+        expected_value = self.generate_redis_answer(id=cmd.job_id, cmd=bash_command, **kwargs)
         self.assertEqual(result, expected_value)
 
+    def test_run_command_is_written_to_redis(self):
```
Comment (DoronCohen) [inline tests/uts/test_client.py]: some of these tests / funcs names are too specific. I think this is more a `test_command_serialization`

File: tests/integration/test_sanity.py (hunk 2 above)
Comment (assafgi) [reply tests/integration/test_sanity.py]: if we have overflow,  the stdout key won't exist, so we don't have what to validate

**Pull request merged**